File: frontend/src/api/weekly.ts
// frontend/src/api/weekly.ts
// 주간 뷰 API 호출 함수

import apiClient from '@/api/client';

export interface MonthlyGoalSubItem {
  title: string;
  card_type: string;
  status: string;
}

export interface MonthlyGoalGroup {
  epic_title: string;
  project_name: string;
  items: MonthlyGoalSubItem[];
}

export interface ActiveProjectItem {
  project_name: string;
  epic_title: string;
}

export interface WeeklyCardItem {
  epic_title: string | null;
  story_title: string | null;
  task_title: string;
  card_type: string;
  due_date: string | null;
  status: string;
  project_name: string;
}

export interface WeeklyViewResponse {
  week_start: string;
  week_end: string;
  month: number;
  monthly_goals: MonthlyGoalGroup[];
  active_projects: ActiveProjectItem[];
  last_week_items: WeeklyCardItem[];
  this_week_items: WeeklyCardItem[];
}

export async function getWeeklyView(weekStart?: string): Promise<WeeklyViewResponse> {
  const params = weekStart ? { week_start: weekStart } : {};
  const response = await apiClient.get<WeeklyViewResponse>('/weekly', { params });
  return response.data;
}

File: backend/app/schemas/weekly.py
# backend/app/schemas/weekly.py
# 주간 뷰 응답 스키마

from datetime import date

from pydantic import BaseModel


class MonthlyGoalSubItem(BaseModel):
    title: str
    card_type: str
    status: str


class MonthlyGoalGroup(BaseModel):
    epic_title: str
    project_name: str
    items: list[MonthlyGoalSubItem]


class ActiveProjectItem(BaseModel):
    project_name: str
    epic_title: str


class WeeklyCardItem(BaseModel):
    epic_title: str | None = None
    story_title: str | None = None
    task_title: str
    card_type: str
    due_date: date | None = None
    status: str
    project_name: str


class WeeklyViewResponse(BaseModel):
    week_start: date
    week_end: date
    month: int
    monthly_goals: list[MonthlyGoalGroup]
    active_projects: list[ActiveProjectItem]
    last_week_items: list[WeeklyCardItem]
    this_week_items: list[WeeklyCardItem]
